File: ProdajaNekretnina.Model/Kupci.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProdajaNekretnina.Model
{
    public class Kupci
    {
        [Key]
        public int KupacId { get; set; }

        public string Ime { get; set; } = null!;

        public string Prezime { get; set; } = null!;

        public DateTime DatumRegistracije { get; set; }

        public string Email { get; set; } = null!;

        public string KorisnickoIme { get; set; } = null!;

        public string LozinkaHash { get; set; } = null!;

        public string LozinkaSalt { get; set; } = null!;

        public bool Status { get; set; }
        public string ClientId { get; set; } 
        public string ClientSecret { get; set; } 
    }
}

File: UI/prodajanekretnina_mobile/lib/models/kupci.dart
import 'package:json_annotation/json_annotation.dart';
import 'dart:ffi';
part 'kupci.g.dart';

@JsonSerializable()
class Kupci {
  int? kupacId;
  String? ime;
  String? prezime;
  String? datumRegistracije;
  String? email;
  String? korisnickoIme;
  bool? status;
  String? clientId;
  String? clientSecret;

  Kupci(
      this.kupacId,
      this.ime,
      this.prezime,
      this.datumRegistracije,
      this.email,
      this.korisnickoIme,
      this.status,
      this.clientId,
      this.clientSecret);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory Kupci.fromJson(Map<String, dynamic> json) => _$KupciFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$KupciToJson(this);
}
